pkg/pipeline/provider.go

```diff
@@ -0,0 +1,190 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var Version = "0.0.1"
+
+func Provider() *schema.Provider {
+	p := &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				DefaultFunc:  schema.MultiEnvDefaultFunc([]string{"PIPELINES_URL", "JFROG_URL"}, "http://localhost:8081"),
+				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
```

Comment:
Please make sure you use the new TF APIs: `DiagValidFunc` or whatever it's called. Our other repos use it